### ISSUE TITLE
implement equality operator for wrapped classes

### ISSUE THREAD

**Issue by pavoljuhas** [diffpy/pyobjcryst#13]
Crystal methods return wrappers to the same C++ instances, but they don't test for identity not equality:

```python
>>>  nacl = loadObjCrystCrystal('NaCl.cif')
>>>  nacl.GetScatterer(0) is nacl.GetScatterer(0)
False
>>>  nacl.GetScatterer(0) == nacl.GetScatterer(0)
False
>>>  a = nacl.GetScatterer(0)
>>>  a.GetScatteringPower() is a.GetScatteringPower()
False
>>>  a.GetScatteringPower() == a.GetScatteringPower()
False
```
TODO

- [ ] fix identity test if possible
- [ ] fix equality operation

**Comment by sbillinge** [comment]
@vincefn please can you look at this.  It is an old one from Pavol.  Do we want to implement this or close the issue as no long needed, or put it to a leter release?  Alll are fine with me.  It may also bave been implemented.  I didn't check.  Maybe @tieqiong could check that.

**Comment by sbillinge** [comment]
@vincefn one more question.  In the rest of diffpy we have moved over stricltly to semantic version numbering (`major.minor.patch`).  Our automated release scripts are based on this.  However, I see you switched at some point to `year.something.patch` or sthg like that.  Because the "type pattern" of your versioning (at least as long as the third integer is there) matches semantic numbering, by chance it would actually still work with our automated release scripts.   Is there a reason you switched and why you like your method?  I would actually be ok with either, though would prefer it personally if we could unify it back to semantic numbering.  But let's discuss.

**Comment by sbillinge** [comment]
@vincefn one more thing.  We are just about ready to ask you to test the "revamped" code.  It is sitting on the `migration` branch in the git repo.  We could try and do a pypi and/or conda pre-release so you could actually test a released version, or else if you like.  These pre-releases are invisible to users unless they go searching for them, so a `pip install pyobjscryst` would still give them `2024.2.1` if we did that.

the objective is that we didn't break anything (or fix anything tbh), just brought everything into line with the rest of diffpy to facilitate maintenance and releases.  @tieqiong has tested the example scripts in the repo and they are all running fine (he had some questions that he put in separate issues e.g., #54) but the best is if you can quickly run some of your scripts and make sure we haven't broken anything.  Then we can merge migration into main and move to a new release.

TL;DR:  
1. please can you test the new version which is curently sitting on the `migration` branch
2. please let us know if you want to test it by building from sources or if you want us to release it to PyPI as a pre-release and test a fully installed version
3. when done, please let us know and we can merge `migration` into `main`

**Comment by sbillinge** [comment]
btw, @vincefn we want to make a new release of `diffpy.cmi` all the way to python 3.13 and so would love to have this version of `pyobjcryst` in that.   Thiw will, for the first time, have `diffpy.cmi` running on windows, as well as latest python, so it is a big deal for us.

**Comment by sbillinge** [comment]
@vincefn if you are interested, we have formalized all our workflows in something we call `scikit-package` (https://scikit-package.github.io/scikit-package/).   I need this because I maintain our code using grad students who cycle through so I want very strict standards.  But we thought it might be useful for the broader scientific software community.
